mux/src/tmux_pty.rs

```diff
@@ -0,0 +1,118 @@
+use crate::{
+    tmux::{RefTmuxRemotePane, TmuxCmdQueue, TmuxDomainState},
+    tmux_commands::SendKeys,
+};
+use flume;
+use portable_pty::{Child, ExitStatus, MasterPty};
+use std::{
+    io::{Read, Write},
+    sync::{Arc, Condvar, Mutex},
+};
+
+pub(crate) struct TmuxReader {
+    rx: flume::Receiver<String>,
+}
+
+impl Read for TmuxReader {
+    fn read(&mut self, mut buf: &mut [u8]) -> std::io::Result<usize> {
+        match self.rx.recv() {
+            Ok(str) => {
+                return buf.write(str.as_bytes());
+            }
+            Err(_) => {
+                return Ok(0);
+            }
+        }
+    }
+}
+
+// A local tmux pane(tab) based on a tmux pty
+#[derive(Debug, Clone)]
+pub(crate) struct TmuxPty {
+    pub domain_id: usize,
+    pub master_pane: RefTmuxRemotePane,
+    pub rx: flume::Receiver<String>,
+    pub cmd_queue: Arc<Mutex<TmuxCmdQueue>>,
+    pub active_lock: Arc<(Mutex<bool>, Condvar)>,
+}
+
+impl Write for TmuxPty {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        let pane_id = {
+            let pane_lock = self.master_pane.lock().unwrap();
+            pane_lock.pane_id
+        };
+        log::info!("pane:{}, content:{:?}", &pane_id, buf);
```

Comment:
nit: info is too high a log level for this :-)

```suggestion
        log::trace!("pane:{}, content:{:?}", &pane_id, buf);
```